lib/Backends/Interpreter/InterpreterNodes.cpp

```diff
@@ -6829,3 +6829,74 @@ void BoundInterpreterFunction::fwdExternalFunctionCallInst(
     glow::ExternalFunctionCallInst const *) {
   LOG(FATAL) << "ExternalFunctionCallInst is not supported yet";
 }
+
+template <typename T, typename InTy>
+void BoundInterpreterFunction::fwdBatchPermutationInstImpl(
+    glow::BatchPermutationInst const *I) {
+  Tensor *inputT = getTensor(I->getInput());
+  Tensor *indicesT = getTensor(I->getIndices());
+  Tensor *outputT = getTensor(I->getOutput());
+
+  auto inputH = inputT->getHandle<InTy>();
+  auto indicesH = indicesT->getHandle<T>();
+  outputT->zero(); // Set elements to zero
+
+  // Calculate number of batch elements, i.e., K need to re-ordered and
+  // corrensponding data slice size.
+  const dim_t batchSize = indicesH.dims()[0];
+  const dim_t K = inputH.size() / inputH.dims()[0];
+  const dim_t elementSize = inputT->getType().getElementSize();
+  const dim_t dataSliceBytes = K * elementSize;
+
+  // For each batch input, calculate the corresponding index in the output
+  // based on indices and copy the data to output tensor.
+  // DistributeFPNPropasals op currently assigns index value as -1 for invalid
+  // ROIs and index value -1 is skipped, since the corresponding index in the
+  // output is filled with zeroes.
+  for (dim_t n = 0; n < batchSize; n++) {
+    if (indicesH.at({n}) != -1) {
```

Comment:
nit: early continue to reduce indentation depth
```suggestion
    if (indicesH.at({n}) == -1) {
      continue;
    }
```